editors/code/src/lsp_ext.ts

```diff
@@ -68,6 +68,37 @@ export const viewItemTree = new lc.RequestType<ViewItemTreeParams, string, void>
     "rust-analyzer/viewItemTree",
 );
 
+export type DiscoverTestParams = { testId?: string | undefined };
+export type RunTestParams = {
+    include?: string[] | undefined;
+    exclude?: string[] | undefined;
+};
+export type TestItem = {
+    id: string;
+    label: string;
+    icon: "package" | "module" | "test";
+    canResolveChildren: boolean;
+    parent?: string | undefined;
+    textDocument?: lc.TextDocumentIdentifier | undefined;
+    range?: lc.Range | undefined;
+    runnable?: Runnable | undefined;
```

Comment:
Ah, do we have `runnable` at the first commit?

The info in Runnable is rich, I believe it already tells `range`, `textDocument`, and `parent`. It would be kind of confused if there is no single truth of source.

Reply:
It doesn't tell `parent`, but it tells `range` and `textDocument`. I will remove the duplicate fields of runnable here.

Reply:
I ended up keeping those fields, since some of them should be kept in sync with the old runnables in order to function correctly without breaking the old runnables, but are not ideal for the test explorer usage. And it makes the implementation for clients that already implement the runnable extension simpler.